src/common/utils/shared.rs

```diff
@@ -43,12 +43,33 @@ pub mod colors {
     pub const BLACK: (u8, u8, u8) = (0, 0, 0);
 }
 
-pub fn hex_to_rgb(hex: &Option<String>) -> (u8, u8, u8) {
+pub fn hex_to_rgb(hex: &Option<String>) -> Option<(u8, u8, u8)> {
```

Comment:
How about removing `Option`? Because `hex_to_rgb` seems to be a util function, I wonder it doesn't have to care about the argument is `Some` or not.

By the way, `Rgb::from_hex_str` returns `Result`, but this function looks like ignoring it with `unwrap`. I think it's OK because `hex_str` is usually a valid color hex in many cases. However, using `expect` might be more helpful for developers to detect errors.

To summarize, I came up with this:

```rust
pub fn hex_to_rgb(hex: &str) -> (u8, u8, u8) {
    let rgb = Rgb::from_hex_str(hex)
        .expect("The passed argument should be valid color hex.")
        .as_tuple();
    (rgb.0 as u8, rgb.1 as u8, rgb.2 as u8)
}
```

Reply:
Thank you, yes that makes a lot of sense. Your solution is very elegant!